### DISCUSSION TITLE
About infinite timer

### DISCUSSION THREAD

**Discussion by 03psb** [ANLAB-KAIST/KENSv3#235] [Project Q&A 3. Internet Layer (IP/Routing)] [answered]
I basically faced same problem as discussed in https://github.com/ANLAB-KAIST/KENSv3/discussions/233
So I tried out one of TA's solution, which is to call cancelTimer in the finalize() part. But in it's signature, the UUID is required. When I looked at the skeleton code, it is not sure how can I get the UUID for my process. How can I get one?

**Comment by Pusnow** [comment, marked as answer]
When you create a timer, `addTimer()` returns a UUID. You can use it to cancel the timer.

However, canceling timers would not solve your problem. The problem would be from routing loops that increase simulation time exponentially.
